### PR TITLE
OpenAPI v3.0 schema validation

### PR DESCRIPTION
**Pull request by jsanda**
This PR is for #34.

### CODE REVIEW

**Comment by jsanda** [comment]
@allamand / @cscetbon the initial validation changes are in place. I have some questions:

Do we want regex validation for `CassandraResources` for cpu and memory?

Do we want to make sure `DataCapacity` is > 0 (if specified)?

Do we want to make sure `MaxPodUnavailable` is > 0 (if specified)?

Do we want to make sure `ImagePullPolicy` has a valid value (this would be done with an enum tag)?

Do we want to make sure `RunAsUser` is > 0 (if specified)?

Something else that I thought could be interesting is that if `ConfigMapName` is set, then make sure that there is a configmap with the specified name. This would have to be done with a validating admission web hook. If there is interest in something like this, I would be happy to do it in a follow up PR.

**Comment by jsanda** [comment]
@allamand / @cscetbon 

Any thoughts on this? And can you run the e2e tests please?

**Comment by allamand** [comment]
Hi John,

CircleCI has removed docker layer caching for free tiers. We need to update circleCI config to remove that. You can wait for my branch to be merged then rebased on it

**Comment by allamand** [comment]
Hi @jsanda, 
> Do we want regex validation for `CassandraResources` for cpu and memory?
I'm not sure how this is done for pod for instance, we should do same thing here

> Do we want to make sure `DataCapacity` is > 0 (if specified)?
This makes sense
 
> Do we want to make sure `MaxPodUnavailable` is > 0 (if specified)?
I wonder If MaxPodUnavailable=0 could be useful in some cases ?

> Do we want to make sure `ImagePullPolicy` has a valid value (this would be done with an enum tag)?
This makes sense

> Do we want to make sure `RunAsUser` is > 0 (if specified)?
This makes sense

> Something else that I thought could be interesting is that if `ConfigMapName` is set, then make sure that there is a configmap with the specified name. This would have to be done with a validating admission web hook. If there is interest in something like this, I would be happy to do it in a follow up PR.
This could be useful also, but not mandatory (or configurable). Four our case we know we won't be able to install some admission webhook in our multi-tenant, this will be forbidden by our admins.
Saying that, this could be useful for folks that are allowed to install some.

**Comment by jsanda** [comment]
@allamand 

I have added additional validations. Is there anything else that should be done in this PR?

**Comment by cscetbon** [comment]
@allamand, please answer john’s question.

**Comment by fdehay** [comment]
Hi @jsanda, does this PR fix #89 like you said in #87 ? Seems yes but not too sure so I prefer asking :)

**Comment by jsanda** [comment]
Hi @fdehay,

I pushed a commit that fixes the naming for the following:

* `CassandraClusterSpec.UnlockNextOperation`
* `PodLastOperation.Name`
* `CassandraLastAction.Name`
* `CassandraClusterStatus.SeedList`

I have not made changes to:

* `CassandraClusterSpec.GCStdout`
* `CPUAndMem.CPU`
* `Topology.DC`

I did not change these because there are different conventions for how to handle abbreviations and acronyms in camel case. I think that the Go identifier names are fine. I am just now sure what the JSON names should be.

File: deploy/crds/db_v1alpha1_cassandracluster_crd.yaml [hunk 1]
```diff
@@ -10,4 +10,313 @@ spec:
...
+          properties:
+            cassandraRackStatus:
+              additionalProperties:
+                properties:
+                  cassandraLastAction:
+                    description: 'CassandraLastAction is the set of Cassandra State
+                      & Actions: Active, Standby..'
+                    properties:
+                      Name:
+                        description: 'Type d''action a effectuer : UpdateVersion,
```
Comment (cscetbon) [inline deploy/crds/db_v1alpha1_cassandracluster_crd.yaml]: @jsanda you're a french speaker now ? ha ha. It should be translated

File: deploy/crds/db_v1alpha1_cassandracluster_crd.yaml [hunk 2]
```diff
@@ -10,4 +10,313 @@ spec:
...
+                        description: List of Racks defined in the Cassandra DC
+                        items:
+                          properties:
+                            labels:
+                              additionalProperties:
+                                type: string
+                              description: Labels used to target Kubernetes nodes
+                              type: object
+                            name:
+                              description: Name of the Rack
```
Comment (cscetbon) [inline deploy/crds/db_v1alpha1_cassandracluster_crd.yaml]: Please ensure that dc and rack names do not contain a dash as it is used internally as a delimiter. See #158

File: pkg/apis/db/v1alpha1/cassandracluster_types.go [hunk 3]
```diff
@@ -682,9 +683,10 @@ type CassandraClusterSpec struct {
 	MaxPodUnavailable int32 `json:"maxPodUnavailable"` //Number of MaxPodUnavailable used in the PDB
 
 	//Very special Flag to hack CassKop reconcile loop - use with really good Care
-	UnlockNextOperation bool `json:"_unlockNextOperation,omitempty"`
+	UnlockNextOperation bool `json:"unlockNextOperation,omitempty"`
 
 	//Define the Capacity for Persistent Volume Claims in the local storage
+	// +kubebuilder:validation:Pattern=^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
```
Comment (cscetbon) [inline pkg/apis/db/v1alpha1/cassandracluster_types.go]: What is that comment ?

File: deploy/crds/db_v1alpha1_cassandracluster_crd.yaml (hunk 1 above)
Comment (jsanda) [reply deploy/crds/db_v1alpha1_cassandracluster_crd.yaml]: rien qu'un peu :)

File: deploy/crds/db_v1alpha1_cassandracluster_crd.yaml (hunk 2 above)
Comment (jsanda) [reply deploy/crds/db_v1alpha1_cassandracluster_crd.yaml]: I added validation regexes

File: pkg/apis/db/v1alpha1/cassandracluster_types.go (hunk 3 above)
Comment (jsanda) [reply pkg/apis/db/v1alpha1/cassandracluster_types.go]: Based on my understanding of the code `DataQuantity` ends up being treated as a `Quantity` just like the `CPUAndMem.CPU` and `CPUAndMem.CPU` fields which have the same regex. See https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L30 for more details.

**Pull request merged**